skt/reporter.py

```diff
@@ -234,6 +234,24 @@ def __init__(self, cfg):
         # multireporting
         self.multi_job_ids = []
 
+    def get_config_local(self):
+        """Retrieve a local config file from the disk."""
+        config_file = "{}/.config".format('workdir')
+        if os.path.isfile(config_file):
+            # The config file
+            with open(config_file, "r") as fileh:
+                return fileh.read()
+
+        return None
+
+    def get_config_remote(self):
+        """Retrieve a config file from a remote location."""
+        response = requests.get(self.cfg.get("cfgurl"))
+        if response.status_code == requests.codes.ok:
+            return response.text
+
+        return None
```

Comment:
Can we have this return None, only if "cfgurl" is not set? Then we'll be able to simply try to get the remote config, and if it fails, get the local one?

Actually, why do we even get the remote config?

Reply:
This function only gets called if `cfg['cfgurl']` is set.

Reply:
Yeah, that's what I noticed too. So I was thinking, make it return None only if `cfgurl` is not set, and make it throw an exception if it failed to retrieve the configuration. Then we'll be able to write this:

    config = self.get_config_remote() or self.get_config_local()

and the `update_mergedata()` function won't need to know anything about `cfgurl`.